### ISSUE TITLE
CarouselView notify when scrolled

### ISSUE THREAD

**Issue by ChasakisD** [xamarin/Xamarin.Forms#4654]
## Summary
In CarouselView, the only way to change the visible item is by using the ```ScrollTo``` method of ```ItemsView```. This is very useful so you can change the visible item(or selectedindex) from the Xamarin.Forms. But, when the user swipes the Abstraction Layer never gets notified. From a little workaround, in Android the ```ItemsViewRenderer``` could add a ```OnScrollListener``` and inside the ```OnScrolled``` by the ```FindFirstVisibleItemPosition``` or ```FindFirstCompletelyVisibleItemPosition``` that exists on both ```LinearLayoutManager``` and ```GridLayoutManager``` the Renderer could notify the Abstraction Layer.

## API Changes
Maybe add a property like ```SelectedIndex``` so the developer will know when the visible item is changed, or maybe just a ScrolledEvent with EventArgs that containes the new Index. This would a CarouselView specific property or could apply to the CollectionView/ItemsView.

## Intended Use Case
This API will be very important when using the ```CarouselView``` with ```SnapPointsType``` to ```MandatorySingle```, so the developer will know when the user moved to the next item. This one can extend the functionality of the CarouselView and it will give us the opportunity of using it as a TabLayout.

EDIT2: On iOS There is the Scrolled Event on the UICollectionViewController. I could make a PR if the proposal gets accepted :)

EDIT: A dummy RVScrollListener which i got a working sample is the following:
The RVScrollListener added OnElementChanged and removed on Dispose, like the following
```cs
_scrollListener = new RVScrollListener(e.NewElement as CustomCarouselView);
AddOnScrollListener(_scrollListener); 
```
```cs
class RVScrollListener : OnScrollListener
{
	private CustomCarouselView _carouselView;

	public RVScrollListener(CustomCarouselView carouselView)
	{
		_carouselView = carouselView;
	}

	public override void OnScrolled(RecyclerView recyclerView, int dx, int dy)
	{
		base.OnScrolled(recyclerView, dx, dy);

		var position = -1;

		var layoutManager = recyclerView.GetLayoutManager();

		switch (layoutManager)
		{
			case GridLayoutManager gridLayoutManager:
				position = gridLayoutManager.FindFirstCompletelyVisibleItemPosition();
				break;
			case LinearLayoutManager linearLayoutManager:
				position = linearLayoutManager.FindFirstCompletelyVisibleItemPosition();
				break;
		}

		if (position != -1)
		{
			_carouselView?.OnScrollChanged(position);
		}
	}
}
```

Of course there must be more checks because the OnScrolled may be called multiple times, if for example the user goes back and forth the same item, so it will fire the event only when the position has been changed.

**Comment by Magendanz** [comment]
Why doesn't CarouselView just inherit from SelectableItemsView?  This would give us both SelectedItem and a SelectionChanged event (as well SelectionChangedCommand).

**Comment by jfversluis** [comment]
With merging #5754 I’m going to close this for now. Should’ve happened automatically, but didn’t ¯\_(ツ)_/¯

**Comment by Magendanz** [comment]
This is not at all fixed with #5745.  We need SelectedItem and a SelectedIndex for MVVM implementations.

**Comment by jfversluis** [comment]
@Magendanz are you sure this is the right issue? This has to do with the scrolling event, not selecting anything.
